staging/traefik/templates/tests/test-configmap.yaml

```diff
@@ -8,9 +8,24 @@ metadata:
     chart: {{ template "traefik.chart" . }}
     heritage: "{{ .Release.Service }}"
     release: "{{ .Release.Name }}"
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "6"
 data:
   run.sh: |-
-    @test "Test Access" {
-      curl -D - http://{{ template "traefik.fullname" . }}/
+    @test "Test HTTP" {
+      curl -D - http://{{ template "traefik.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local./
+    }
+    @test "Test HTTPS" {
+      curl -k -vvv -D - https://{{ template "traefik.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local./
+    }
+    @test "Test Default Certificate" {
+      echo -n | openssl s_client -connect {{ template "traefik.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local.:443 2> /dev/null | openssl x509 -noout -subject | grep -q "traefik.localhost.localdomain"
+    }
+    @test "Test Default Certificate SNI" {
+      echo -n | openssl s_client -servername custom.whalesdolphines.tld -connect {{ template "traefik.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local.:443 2> /dev/null | openssl x509 -noout -text | grep -q "custom.whalesdolphines.tld"
```

Comment:
lol "whalesdolphines", sounds like someone is looking forward to an oceanic retreat! :beach_umbrella: